azure/services/aso/aso.go

```diff
@@ -100,6 +101,12 @@ func (s *Service) CreateOrUpdateResource(ctx context.Context, spec azure.ASOReso
 		var readyErr error
 		if cond := conds[i]; cond.Status != metav1.ConditionTrue {
 			switch {
+			case cond.Reason == conditions.ReasonAzureResourceNotFound.Name &&
```

Comment:
So this condition describes when we have a pre-existing ASO resource that doesn't actually have an Azure resource behind it?

Reply:
Yes, but only "pre-existing" in the sense that CAPZ created in the very near past. Pre-existing ASO resources not created by CAPZ should never reach here based on the `ownedByCluster` check above. In cases when the resource already exists in Azure but outside of CAPZ's ASO, those resources should never transition to this "AzureResourceNotFound" state.